job-dsls/jobs/tools_provisioning.groovy

```diff
@@ -0,0 +1,428 @@
+            preBuildCleanup()
+
+            // Loads file provider
+            configFileProvider {
+                managedFiles {
+                    configFile {
+                        fileId('jenkins.plugins.openstack.compute.UserDataConfig.1454073256508')
+                        targetLocation('\$WORKSPACE/rhel-cloud-init')
+                        variable('CLOUD_INIT')
+                    }
+                    configFile {
+                        fileId('openstack-upshift-rc-file')
+                        targetLocation('\$WORKSPACE/openrc.sh')
+                        variable('OPENRC_FILE')
+                    }
+                }
+            }
+
+            if ( jobAbr == 'DockerReg') {
+                credentialsBinding {
+                    // Sets one variable to the username and one variable to the password given in the credentials.
+                    usernamePassword('DOCKERHUB_USER','DOCKERHUB_PASSWORD', 'kie-ci-dockerhub')
+                }
+            }
+
+            environmentVariables {
+                // Adds an environment variable to the build.
+                env("INVENTORY_FILE", "\$WORKSPACE/cekit-cacher-inventory.txt")
+                env("MACHINES_FILE","\$WORKSPACE/machines.txt")
+                env("IP_FILE","\$WORKSPACE/ips.txt")
+            }
+
+            // Injects passwords as environment variables into the job.
+            injectPasswords {
+                // Injects global passwords provided by Jenkins configuration.
+                injectGlobalPasswords(true)
+                // Masks passwords provided by build parameters.
+                maskPasswordParameters(true)
+            }
+        }
+
+        // Adds custom properties to the job.
+        properties {
+
+            // Allows to configure job ownership.
+            ownership {
+
+                // Sets the name of the primary owner of the job.
+                primaryOwnerId("mbiarnes")
+
+                // Adds additional users, who have ownership privileges.
+                coOwnerIds("anstephe", "mnovotny", "almorale")
+            }
+        }
+
+        // Adds publishrs o job
+        publishers {
+            // Archives artifacts with each build.
+            archiveArtifacts('*.txt')
+            wsCleanup()
+        }
+
+        // Adds step that executes a shell script
+        steps {
+            shell(getScripts(jobAbr))
+            shell(getScripts_II(jobAbr))
+        }
+    }
+}
+
+// def of variables
+String getDescription(String Abr){
+    switch(Abr) {
+        case "VerdaccioServ":
+            return "The job for provisioning verdaccio service node."
+        case "DockerReg":
+            return "The job for provisioning local Docker Registry cache node."
+        case "SmeeClient":
+            return "The job for provisioning smee client node."
+        default:
+            return "The job for provisioning cekit-cacher node."
+    }
+}
+
+String getScripts(String Abr) {
```

Comment:
If it's possible I would move all static scripting/contents to the external resources folder as we did in this PR: https://github.com/kiegroup/kie-jenkins-scripts/pull/930/files#diff-a8ed59266e5fab543f3c05be2112ae04f816ffc5817102002e7dab423d8c8f6fR6

You can drop any script in the `src/main/resources` folder and load it calling `this.getClass().getResource()`.
If you name the files after the jobname you can load them automatically without any additional logic. Something like creating a file `src/main/resources/job-scripts/tools_provisioning_verdaccioServ.sh` and loading it with 

```this.getClass().getResource("/job-scripts/tools_provisioning_" + jobName + ".sh")```